File: examples/java-ddd/application/src/test/java/org/myddd/java/AbstractTest.java
package org.myddd.java;

import org.junit.jupiter.api.BeforeEach;
import org.myddd.domain.InstanceFactory;
import org.myddd.ioc.spring.SpringInstanceProvider;
import org.myddd.java.api.dto.UserDTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.ComponentScan;

import java.util.UUID;

@SpringBootTest(classes = AbstractTest.class)

@SpringBootApplication
@ComponentScan(basePackages = {"org.myddd","org.myddd.java"})
@EntityScan(basePackages = {"org.myddd","org.myddd.java"})
public abstract class AbstractTest {

    @Autowired
    private ApplicationContext applicationContext;

    @BeforeEach
    public void beforeClass(){
        InstanceFactory.setInstanceProvider(SpringInstanceProvider.createInstance(applicationContext));
    }

    public String randomString(){
        return UUID.randomUUID().toString().replace("-","");
    }


    public UserDTO randomUserDTO(){
        var user = new UserDTO();
        user.setUserId(randomString());
        user.setName(randomString());
        user.setPhone(randomString());
        user.setEmail(randomString()+"@myddd.org");
        return user;
    }


}

File: samples/kotlin-ddd/application/src/test/kotlin/org/myddd/kotlin/AbstractTest.kt
package org.myddd.kotlin

import org.junit.jupiter.api.BeforeEach
import org.myddd.domain.InstanceFactory
import org.myddd.ioc.spring.SpringInstanceProvider
import org.myddd.kotlin.api.dto.UserDTO
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.domain.EntityScan
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.ApplicationContext
import org.springframework.context.annotation.ComponentScan
import java.util.*
import javax.inject.Inject

@SpringBootTest(classes = [AbstractTest::class])
@SpringBootApplication
@ComponentScan(basePackages = ["org.myddd"])
@EntityScan(basePackages = ["org.myddd"])
abstract class AbstractTest {

    @Inject
    lateinit var applicationContext:ApplicationContext

    @BeforeEach
    fun beforeEach(){
        InstanceFactory.setInstanceProvider(SpringInstanceProvider.createInstance(applicationContext))
    }

    companion object {
        fun randomString():String {
            return UUID.randomUUID().toString().replace("-","")
        }
    }

    fun randomUserDTO():UserDTO {
        return UserDTO(userId = randomString(), name = randomString(), phone = "18620501006", email = "${randomString()}@taoofcoding.tech")
    }
}